src/modules/orders.ts

```diff
@@ -5,35 +5,48 @@ import { Order } from "../../generated/schema"
 
 export namespace orders {
 
-    export function invalidateOrder(orderId: string): void {
+    export function invalidateOrder(orderId: string, timestamp: BigInt): Order {
 
         let order = Order.load(orderId)
 
         if (!order) {
             order = new Order(orderId)
-            log.warning('Order {} was not found. It was created for being invalidated', [orderId])
+            log.info('Order {} was not found. It was created for being invalidated', [orderId])
         }
 
         order.isValid = false
-        order.save()
+        order.invalidateTimestamp = timestamp
+
+        return order as Order
+    }
+
+    export function preSig(orderId: string, owner: string, timestamp: BigInt): Order {
```

Comment:
nitpick: I wouldn't shorten into `preSign`. Probably i would pick `setPresignature` to make it closer to the contract

Reply:
changed